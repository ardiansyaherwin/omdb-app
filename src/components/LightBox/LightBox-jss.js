const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: "absolute",
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: 'white',
    background: "rgba(0,0,0,.9)",
  },
  dialogContent: {
    padding: "0 !important",
    border: 0,
  },
  dialogContentImage: {
    display: "block",
    width: "100%",
  },
});

export default styles;
