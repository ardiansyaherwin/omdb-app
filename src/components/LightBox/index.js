import React from "react";
import { PropTypes } from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import {
  Dialog,
  DialogContent,
  IconButton,
} from "@material-ui/core";
import { Close } from "@material-ui/icons";

import styles from "./LightBox-jss";

const LightBox = (props) => {
  const { source, classes, showDialog, onClose } = props;
  const handleClose = () => {
    onClose()
  };

  return (
    <Dialog
      onClose={handleClose}
      aria-labelledby="customized-dialog-title"
      open={showDialog}
    >
      <DialogContent dividers className={classes.dialogContent}>
        <img src={source} className={classes.dialogContentImage} alt="" />
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={handleClose}
        >
          <Close />
        </IconButton>
      </DialogContent>
    </Dialog>
  );
};

LightBox.propTypes = {
  source: PropTypes.string.isRequired,
  showDialog: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default withStyles(styles)(LightBox);
