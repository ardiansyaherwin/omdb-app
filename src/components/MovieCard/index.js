import React, { useState } from "react";
import { PropTypes } from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { Link } from "react-router-dom"
import {
  Typography,
  Card,
  CardActionArea,
  CardMedia,
  CardContent,
  CardActions,
  Button,
} from "@material-ui/core";
import { LightBox } from "../index";

import styles from "./MovieCard-jss";

const MovieCard = (props) => {
  const { 
    poster,
    title,
    type,
    year,
    imdbId,
    classes,
    fullPoster,
    lightbox
  } = props;
  
  const [openLightbox, setOpenLightbox] = useState(false);
  
  const MovieImageCard = () => {
    return <CardMedia image={poster} title={title} style={{ minHeight: 300 }} />
  }

  return (
    <Card className={classes.root}>
      {lightbox ? (
        <>
          <CardActionArea
            onClick={() => setOpenLightbox(true)}
          >
            <MovieImageCard />
          </CardActionArea>
          <LightBox
              source={fullPoster ? fullPoster : poster}
              showDialog={openLightbox}
              onClose={() => setOpenLightbox(false)}
            />
          </>
      ) : (
        <MovieImageCard />
      )}

      <CardContent>
        <Typography color="textSecondary">{type}</Typography>
        <Typography gutterBottom variant="h5" component="h2">
          {title}
        </Typography>
        <Typography color="textSecondary">{year}</Typography>
      </CardContent>
      <CardActions>
        <Button
          size="small"
          color="primary"
          component={Link}
          to={`/movies/${imdbId}`}
        >
          Details
        </Button>
      </CardActions>
    </Card>
  );
};

MovieCard.propTypes = {
  classes: PropTypes.object.isRequired,
  lightbox: PropTypes.bool,
  poster: PropTypes.string.isRequired,
  fullPoster: PropTypes.string,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  imdbId: PropTypes.string.isRequired,
};

MovieCard.defaultProps = {
  lightbox: false,
  fullPoster: null,
};


export default withStyles(styles)(MovieCard);
