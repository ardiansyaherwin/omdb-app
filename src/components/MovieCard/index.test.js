import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter, Link } from "react-router-dom";
import MovieCard from "./index";
import LightBox from "../LightBox"
import { CardActionArea } from "@material-ui/core";
import { unwrap } from "@material-ui/core/test-utils";

const MovieCardNaked = unwrap(MovieCard)
const requiredProps = {
  classes: {},
  poster: 'http://image.com/image.jpg',
  title: 'Movie Title',
  type: 'Movie',
  year: '2020',
  imdbId: '1234',
};

const defaultProps = {
  lightbox: false,
  fullPoster: null,
};


describe('<SearchBar />', () => {
  it("should render as expected", () => {
    const wrapper = shallow(<MovieCard {...requiredProps} {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have props:poster value which suffixed by .jpg, .jpeg or .png", () => {
    const isValidImage = requiredProps.poster.match(/\.(jpeg|jpg|)$/) != null;
    expect(isValidImage).toBeTruthy();
  });

  it("should render a link button with props:to equal to /movies/:imdbId", () => {
    const wrapper = mount(
      <MemoryRouter>
        <MovieCard {...requiredProps} {...defaultProps} />
      </MemoryRouter>
    );
    expect(wrapper.find(Link).props().to).toBe(`/movies/${requiredProps.imdbId}`)
  });

  it("should render <CardActionArea /> if props:lightbox is set to TRUE", () => {
    const wrapper = shallow(<MovieCardNaked {...requiredProps} lightbox />);
    expect(wrapper.find(CardActionArea).exists()).toBeTruthy();
  })

  it("should render <LightBox /> if props:lightbox is set to TRUE", () => {
    const wrapper = shallow(<MovieCardNaked {...requiredProps} lightbox />);
    expect(wrapper.find(LightBox).exists()).toBeTruthy();
  });

  it("should set <LightBox /> props:showDialog to TRUE upon clicking on <CardActionArea />", () => {
    const wrapper = shallow(<MovieCardNaked {...requiredProps} lightbox />)
    wrapper
      .find(CardActionArea)
      .simulate("click")
    
    wrapper.update()
    expect(wrapper.find(LightBox).props().showDialog).toBe(true)
  });
})
