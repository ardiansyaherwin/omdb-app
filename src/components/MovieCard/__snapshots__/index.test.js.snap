// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render as expected 1`] = `
<MovieCard
  classes={
    Object {
      "expand": "MovieCard-expand-3",
      "expandOpen": "MovieCard-expandOpen-4",
      "media": "MovieCard-media-2",
      "root": "MovieCard-root-1",
    }
  }
  fullPoster={null}
  imdbId="1234"
  lightbox={false}
  poster="http://image.com/image.jpg"
  title="Movie Title"
  type="Movie"
  year="2020"
/>
`;
